src/nagini_translation/translators/program.py

```diff
@@ -667,6 +674,316 @@ def create_may_set_predicate(self, ctx: Context) -> 'silver.ast.Predicate':
                                             pos, info)
         return may_set_pred
 
+    def _get_adt_cons_param_decl(self, cons: PythonClass, adt_name: str,
+                                 adt_type: DomainType, pos: Position,
+                                 info: Info) -> List[VarDecl]:
+        """
+        Returns the parameters of the ADT constructor as a list of variable
+        declarations.
+        """
+        arguments = []
+        for arg_name, arg_type in cons.fields.items():
+            argument_type = adt_type if arg_type.type.name == adt_name else self.viper.Ref
+            argument = self.viper.LocalVarDecl(arg_name, argument_type, pos, info)
+            arguments.append(argument)
+        return arguments
+
+    def _get_adt_cons_params(self, cons: PythonClass, adt_name: str,
+                             adt_type: DomainType, pos: Position,
+                             info: Info) -> List[Var]:
+        """
+        Returns the parameters of the ADT constructor as a list of variables.
+        """
+        arguments = []
+        for arg_name, arg_type in cons.fields.items():
+            argument_type = adt_type if arg_type.type.name == adt_name else self.viper.Ref
+            argument = self.viper.LocalVar(arg_name, argument_type, pos, info)
+            arguments.append(argument)
+        return arguments
+
+    def _conjoin(self, eqs: List[Expr], pos: Position, info: Info) -> Expr:
```

Comment:
The conjoin and disjoin methods are probably super useful in general; I think I brainlessly reimplemented this exact functionality in a million places instead of making helper methods like these. We should remember that they exist here and might want to move them to the CommonTranslator if it turns out we can use them elsewhere.